### PR TITLE
New Roboticist and RD traitor mech: Justice.

### PR DESCRIPTION
**Pull request by Xackii**
## About The Pull Request
Video:
https://youtu.be/QwJfO_lbLK0
Text:
Add new traitor mech Justice as Roboticist and RD job restricted item:
Cost 20 tc
Health 175
Damage 60
Only melee attacks(can't attach range weapons(can't attach anything in his arms at all))
Block chanse 60
Speed - in safty mode 2.5 in combat mode 4.5
Attack effects:
   In addition to damage, it has a 50 percent chance of cutting off an arm or leg (does not put you to sleep or give a knockdown 
  effect when hit like regular combat mechs do).
    If a target in crit or stun makes a finishing attack instead of a regular attack, the mech says an epic phrase and after a second cuts off the target's head by dashing through the target.

Special actions(can't be used if mech in safty mode):

Invisability - Makes the mech invisible for 20 seconds. If the mech is pushed or collides with someone or if the pilot presses the invisibility button again, the mech will no longer be invisible and will go into a 5 second cooldown (the same will happen after 20 seconds). If you attack in invisibility, then instead of a normal attack the mech will make an AoE 3x3 attack 1 second after pressing (the attack will be accompanied by a sound and visual effect and for a second the mech will not be able to move, which gives enemies time to run away from the attack). Upon completion AoE attack, everyone( except peoples in crit and in stunned condition) in a 3x3 radius will receive 35 damage and a 25 chance of having one of their limbs cut off.

Charge Attack - Mech makes a dash towards the cursor at a distance of up to 7 cells, breaking all obstacles in its path. If the obstacle is a living person (and he is not in crit and is not stunned) then he will receive 35 damage and a 25 chance of cutting off a limb. Cooldown 5 Seconds.

Also made the sounds of mech melee attacks to vars so that it would be more convenient to change them.
Also fix a runtime error where if you set something in the mech's max_equip_by_category to 0, the mech UI panel doesn't load.

## Why It's Good For The Game

Adds a tritor item unique to the roboticist and rd (similar to chap his grace or a mime mech) with fun and interesting mechanics. This is a new mech with new unique abilities that turns the battle with you into a boss fight. 

:cl:
add: New traitor mech: Justice. For Roboticist and RD.
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
I will be honest I don't understand how any of this concept is relevant to lawyers

**Comment by Rex9001** [comment]
Pretty cool melee mech, but why is it a lawyer traitor item?

**Comment by Xackii** [comment]
In general, the concept is something like a fighter for justice. When corrupt officers don't listen to the law, a lawyer should show them where they belong

**Comment by mc-oofert** [comment]
sanabi

**Comment by Jacquerel** [comment]
It isn't even wielding a gavel...

**Comment by paganiy** [comment]
— Ah ah. HoS! Do you read me?
— Yep. Loud and clear.
— HoS! HoS! It's so cool! There's a **giant mech** in the brig!
— A mech?
— Yes! **A giant mech!** It's so cool! Did you put this here?
— I... have never seen that thing before.
— Oh, it's says something here. **Sa... Nabi...** What's a **Sanabi**?
— Officer... **Listen to head**. **Get out of the brig right now**. **RIGHT NOW!**
— Huh? What? I can't hear you. Teehee. HoS! This mech is movi!@#%
— **NOW GET OUT OF THE BRIG**. Do you hear me? Hello?

**Comment by Xackii** [comment]
> — Ah ah. HoS! Do you read me? — Yep. Loud and clear. — HoS! HoS! It's so cool! There's a **giant mech** in the brig! — A mech? — Yes! **A giant mech!** It's so cool! Did you put this here? — I... have never seen that thing before. — Oh, it's says something here. **Sa... Nabi...** What's a **Sanabi**? — Officer... **Listen to head**. **Get out of the brig right now**. **RIGHT NOW!** — Huh? What? I can't hear you. Teehee. HoS! This mech is movi!@#% — **NOW GET OUT OF THE BRIG**. Do you hear me? Hello?


students

**Comment by Profakos** [comment]
Invisibility is already taken up by the Recitence... and Justice isn't invisible, its blind!

**Comment by Admiral-Gigaton** [comment]
this is really funny, giving a mech to the lawyer of all things, but...

how in the hell does this connect to a LAWYER? sure i get the "IM BATMAN, JUSTICE" part of the gimmick, but i mean... melee only? charging? high delimb? going invis?? fun cool mechanics but again how does this connect to the lawyer, and not like the CHEF, being a melee centric combatant (if in combat, anyways) having CQC and butcher knives.

still, funny giving a mech to a lawyer of all jobs, just doesnt make sense

**Comment by Jacquerel** [comment]
I will be honest I'm struggling to imagine a world where we put this into the game.
I'm extremely iffy on putting any more mechs in the uplink for _anyone_.
I cannot see how we would give one to the lawyer.

IDK maybe you can find another maintainer to argue your corner who is seeing something I am missing but, I would not hold your breath.

**Comment by Xackii** [comment]
> I will be honest I'm struggling to imagine a world where we put this into the game. I'm extremely iffy on putting any more mechs in the uplink for _anyone_. I cannot see how we would give one to the lawyer.
> 
> I can see a world where we give some kind of item to laywers but I can't imagine how it would be this one. This item has basically nothing to do with laywers at all. It's a huge non-sequitur.
> 
> IDK maybe you can find another maintainer to argue your corner who is seeing something I am missing but, I would not hold your breath.

This mech is from the game sanabi (something like that). I don’t know who else might be suitable for a mech that, according to the lore, should fight for justice (an assistant?). Initially, I thought about adding this mech to nuki ops, but I wasn’t sure if this mech was suitable for them too.

**Comment by Jacquerel** [comment]
Telling me that this is a reference to a video game unfortunately is not a very appealing argument to me; I don't care about the lore of some other video game, I only care about what you are adding to space station 13.

**Comment by Admiral-Gigaton** [comment]
> ss13 is the embodiment of references from other games

bruh their argument still stands, you need some real reasoning for this to work

**Comment by Xackii** [comment]
> > ss13 is the embodiment of references from other games
> 
> bruh their argument still stands, you need some real reasoning for this to work

I don’t know what arguments can be inserted here. The complaint is that he doesn't think mech looks like mech for a lawyer, and neither do you. But what should I do about it? The reason why they want to close it is that, according to the maintainer, the mech does not look like the mech of a lawyer. They didn’t tell me what I should do to prevent the PR from being closed. What should I answer? So I think mech is suitable for a lawyer? This is what I think, and not the maintainer who thinks so. If the maintainer has ideas that I can make to make the mech more suitable for how he sees the lawyer's mech, then I'm listening.

**Comment by PunishedHideo** [comment]
Sanabi deniers should perish

**Comment by Xackii** [comment]
if not a lawyer, then i you can add this mech to ops. 
all their mechs are an improved version of the mechs that are on the station and this mech provides unique mechanics

**Comment by Admiral-Gigaton** [comment]
> if not a lawyer, then i you can add this mech to ops. all their mechs are an improved version of the mechs that are on the station and this mech provides unique mechanics

This is true, ops don't have much else for mechs besides the meta, that being the mauler. Even then, face to face combat tends to do better (speed and shitty mech wounds that turn it into a spinning heap of shit that suffocates and burns you) and it would probably fit better for ops than... the lawyer

**Comment by tralezab** [comment]
Alright settle down fellas... it's clear there is a large fundamental issue with how you get this mech (mechs just do not make good job specific traitor items, in fact mechs do not make good traitor items period, i'm with jac on that) but otherwise the mech itself considering nothing else is very cool and so i ask: is there another implementation for how you get the mech that would be more agreeable? some kind of unique traitor sabotage like how antagonists can hack meteor satellites to get the dark matteor? that's how i lock cool things behind hard work 💪

**Comment by tralezab** [comment]
> The Shogun is a lawyer of his time

this post goes insanely hard for a github comment and i think it properly stresses my point that we should not throw the baby out with the bathwater

**Comment by Xackii** [comment]
> Alright settle down fellas... it's clear there is a large fundamental issue with how you get this mech (mechs just do not make good job specific traitor items, in fact mechs do not make good traitor items period, i'm with jac on that) but otherwise the mech itself considering nothing else is very cool and so i ask: is there another implementation for how you get the mech that would be more agreeable? some kind of unique traitor sabotage like how antagonists can hack meteor satellites to get the dark matteor? that's how i lock cool things behind hard work 💪


I just don’t know where. Meteor shields and dark meteor sound logical. how to make it logical for mech.

**Comment by Sakamoto4ka** [comment]
> > The Shogun is a lawyer of his time
> 
> no one cares

you just need to make this pr after someone added judge mech with big hammer. Its simple

**Comment by Xackii** [comment]
> > > The Shogun is a lawyer of his time
> > 
> > 
> > no one cares
> 
> you just need to make this pr after someone added judge mech with big hammer. Its simple


then my pr will not be relevant and they will tell me that I don’t understand anything about art

**Comment by Rex9001** [comment]
What if you lock the parts behind emagging an exofab?

**Comment by Xackii** [comment]
> What if you lock the parts behind emagging an exofab?

it's possible

**Comment by Profakos** [comment]
The mech should also have different sprites, direct references to other games do not go down well anymore, and they neither should.

**Comment by Xackii** [comment]
> The mech should also have different sprites, direct references to other games do not go down well anymore, and they neither should.

Well, this is not an exact copy of the mech, but that’s how I tried to transfer this idea to SS13

**Comment by Jackal-boop** [comment]
how is this related to the lawyer in any way. job tot items should be in line with their occupation. EG doctors get evil surguries.

**Comment by SomeoneYouProbablyKnow** [comment]
Everyone's focused on how it's a mech and how it's not lawyerly but i'm just curious if it's any good or any fun.

As is you're gonna cut off a few heads and get lasered to death, yea you got block chance and invisibility but what are you going to do about to dudes running around you shooting you? You just die and there go your 20 TC.
And if you don't die... you just stab people untill they die. Just a more complicated version of a sword.

It does LOOK cool as heck even if i have no idea what the reference is

**Comment by projectkepler-ru** [comment]
ok first and foremost
you're talented
and this is cool
but how is this a lawyer thing???

**Comment by moocowswag** [comment]
It should be liberty prime mech

**Comment by Xackii** [comment]
> I will be honest I don't understand how any of this concept is relevant to lawyers

Ok i made it roboticist and rd job restricted item instead of lawyer

**Comment by Jacquerel** [comment]
I still don't know if I want another "buy a full mech" in the uplink but I'll remove the tag in case another maintainer is interested in this PR.

**Comment by Xackii** [comment]
I'm still open to ideas on where or how to put this mech. The idea to add it to the lawyer was also so that this mech would not be bought every round

**Comment by projectkepler-ru** [comment]
> I'm still open to ideas on where or how to put this mech. The idea to add it to the lawyer was also so that this mech would not be bought every round

perfectly reasonable 
dont get me wrong this is really cool and well coded, it was ust seemingly abit excessive at first lol

**Comment by TheNobodyofimportance** [comment]
How about this for building the mech, make it an upgrade like the ripley MK.2 that is applied to a durand/gygax/whatever. Thus there needs to be a measure of investment and work into gaining the mech rather than just popping it down inside the clown room [He slipped me :(]

Whatever mech you need to build is at your discretion of course but this seems like an elegant way to build it.

**Comment by Xackii** [comment]
> How about this for building the mech: make it an upgrade like the ripley MK.2 that is applied to a durand/gygax/whatever. Thus there needs to be a measure of investment and work into gaining the mech rather than just popping it down inside the clown room [He slipped me :(]
> 
> Whatever mech you need to build is at your discretion of course but this seems like an elegant way to build it.

like some kind of tritor thing that attaches to an existing mech and turns it into a new one?

**Comment by scrunklershrungle932** [comment]
> Alright settle down fellas... it's clear there is a large fundamental issue with how you get this mech (mechs just do not make good job specific traitor items, in fact mechs do not make good traitor items period, i'm with jac on that) but otherwise the mech itself considering nothing else is very cool and so i ask: is there another implementation for how you get the mech that would be more agreeable? some kind of unique traitor sabotage like how antagonists can hack meteor satellites to get the dark matteor? that's how i lock cool things behind hard work 💪

Not related to the mech or anything, but I didn't know hacking meteor satellites is a thing that could be done.  What's dark matteor or the dark matteor?

**Comment by Blacklist897** [comment]
> > Alright settle down fellas... it's clear there is a large fundamental issue with how you get this mech (mechs just do not make good job specific traitor items, in fact mechs do not make good traitor items period, i'm with jac on that) but otherwise the mech itself considering nothing else is very cool and so i ask: is there another implementation for how you get the mech that would be more agreeable? some kind of unique traitor sabotage like how antagonists can hack meteor satellites to get the dark matteor? that's how i lock cool things behind hard work 💪
> 
> Not related to the mech or anything, but I didn't know hacking meteor satellites is a thing that could be done. What's dark matteor or the dark matteor?

its from a traitor final objective or if 8 meteor sats gets emagged, its basically a singlo that chases the closest living thing

**Comment by Admiral-Gigaton** [comment]
FINALLY, this makes more sense as a PR

awesome, great job, that was really the only reason people shat on it, love the idea, great execution, godspeed

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:171]:
```suggestion
	return ..()
```

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:180]: autodoc comments

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:198]: `if(charge_action?.on)`

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:211]: autodoc

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:216]: autodoc

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:231]: autodoc

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:254]: autodoc

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:265]: comments

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:289]: same

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:369]: comments

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:134]: pls just early return !heavy, else is bad

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:69]: `for(var/mob/living/occupant in occupants)`?

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:103]: early return this

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:105]: `charge_action?.on`

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:163]: `stealth_action?.on`

File: code/modules/vehicles/mecha/combat/justice.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/combat/justice.dm:307]: autodoc comment

File: code/modules/vehicles/mecha/mecha_ui.dm
Comment (jlsnow301) [inline code/modules/vehicles/mecha/mecha_ui.dm:124]: just continue if you can so we don't have to else block this

**Review by jlsnow301 — CHANGES_REQUESTED** [review]
Sprites are sick, need some tlc on the code though,
[Autodoc](https://github.com/tgstation/tgstation/blob/master/.github/guides/AUTODOC.md)
[Early returns](https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-early-returns)
&& optional chaining will help your code quite a bit.

**Comment by Xackii** [comment]
icons moment

**Comment by Xackii** [comment]
> Sprites are sick, need some tlc on the code though, [Autodoc](https://github.com/tgstation/tgstation/blob/master/.github/guides/AUTODOC.md) [Early returns](https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-early-returns) && optional chaining will help your code quite a bit.

A strange problem with icons associated with the new pr for raptors which added a couple of icons to the effects folder. I updated the build and added my icon again, but the merge conflict is not removed. Do I need to reopen pr or what? Also make autodoc and early returns

**Comment by Xackii** [comment]
ok, AnturK helped with solving the problem

**Comment by Xackii** [comment]
3 weeks later

**Comment by Deultus** [comment]
The next enhancement of stun batons will be integrated 10 times faster than this

**Comment by Orchian** [comment]
Please tell me that decapitation didn't delete the brain... Other than that, pretty neat idea.

**Comment by LemonInTheDark** [comment]
@tralezab @jlsnow301 do one of you want this pr as is? (I ask because of jacqu's concerns and the inaction)

**Comment by jlsnow301** [comment]
> @tralezab @jlsnow301 do one of you want this pr as is? (I ask because of jacqu's concerns and the inaction)

i'm a bit biased against mechs, probably not the best person to ask. yes, the sprites are great, but i just don't like mechs enough to want to add anything to them

**Comment by LemonInTheDark** [comment]
> i'm a bit biased against mechs, probably not the best person to ask. yes, the sprites are great, but i just don't like mechs enough to want to add anything to them

~~why did you review it then ahhhh~~

**Comment by jlsnow301** [comment]
> ~why did you review it then ahhhh~

Valid, my apologies, I should have stated directly. I would not block this PR but I have no personal interest in changing mechs (outside of removal)

**Comment by Jacquerel** [comment]
Despite the tag being removed it doesn't seem like there's any interest in merging this and I would frankly usually just rather science staff build mechs instead of buying a fully-made one from an uplink. You can reopen this if you find a maintainer who actually really wants it in the game.

**Comment by LemonInTheDark** [comment]
If you are interested in perusing this further, I would recommend adding a more interesting source for the mech. We don't want people (especially science) just buying something they could make. @tralezab (armhulen on discord) has volenteered to have opinions on this topic, you should reach out to him for more info

image

**Comment by Xackii** [comment]
maybe add mech to nukers?

**Comment by LemonInTheDark** [comment]
don't they already have two? would it fill an interesting role?

**Comment by Xackii** [comment]
> don't they already have two? would it fill an interesting role?

those two are heavy and long-range. This one is more stealth and melee

**Comment by Orchian** [comment]
It doesn't really seem like a mech for standard crew is the issue I see with having it be built as opposed to 'bought.' Maybe having a parts kit intended to be used on a different, existing chassis could be a feasible solution if the issue is simply, not wanting the mech to appear out of thin air?

It seems worthwhile to me to follow up with Tralezab on this one, at least to understand what their gripes are with the current idea in better detail.

**Pull request closed without merging**
